internal/locate/region_cache.go

```diff
@@ -2783,50 +2785,85 @@ func (s *Store) requestLivenessAndStartHealthCheckLoopIfNeeded(bo *retry.Backoff
 	// It may be already started by another thread.
 	if atomic.CompareAndSwapUint32(&s.livenessState, uint32(reachable), uint32(liveness)) {
 		s.unreachableSince = time.Now()
-		go s.checkUntilHealth(c)
+		reResolveInterval := 30 * time.Second
```

Comment:
Better to make it a constant value.

Reply:
The rawkv tests use failpoint to set it shorter.

Reply:
how about 

```go
var DefReResolveInterval = 30 * time.Second // global scope.

func (s *Store) checkUntilHealth(c *RegionCache, liveness livenessState){
  reResolveInterval = DefReResolveInterval
  ...
}
```

and if test needs a shorter reResolveInterval, change `DefReResolveInterval` directly, no need to use the following failpoint.

Reply:
> how about
> 
> ```go
> var DefReResolveInterval = 30 * time.Second // global scope.
> 
> func (s *Store) checkUntilHealth(c *RegionCache, liveness livenessState){
>   reResolveInterval = DefReResolveInterval
>   ...
> }
> ```
> 
> and if test needs a shorter reResolveInterval, change `DefReResolveInterval` directly, no need to use the following failpoint.

Personally I do not prefer that. The global var is required to be handled carefully to avoid data race in ut. I've struggled with `SetRegionCacheTTLSec` in https://github.com/tikv/client-go/pull/1122 for about an hour.